spec/console_spec.rb

```diff
@@ -0,0 +1,131 @@
+        allow(view_module).to receive(:matrix)
+        allow(view_module).to receive(:obtain_new_game).and_return('n')
+      end
+
+      context 'when user win return win message' do
+        before do
+          allow(view_module).to receive(:obtain_save).and_return('n')
+          allow(view_module).to receive(:obtain_guess).and_return('1111')
+          allow_any_instance_of(Codebraker::Game).to receive(:win?).and_return(true, false)
+        end
+
+        it { expect(view_module).to receive(:win) }
+      end
+
+      context 'when user lose return lose message' do
+        before { allow_any_instance_of(Codebraker::Game).to receive(:lose?).and_return(true, false) }
+
+        it { expect(view_module).to receive(:loss) }
+      end
+    end
+
+    context 'when error' do
+      before { allow(view_module).to receive(:fetch_user_input).and_return('start', 'exit') }
+
+      context 'when guess is not correct output error message from view' do
+        before do
+          allow(view_module).to receive(:obtain_name).and_return(name)
+          allow(view_module).to receive(:obtain_difficulty).and_return(difficulty)
+          allow(view_module).to receive(:guess_input_error)
+          allow(view_module).to receive(:obtain_guess).and_return('654123', 'exit')
+        end
+
+        it { expect(view_module).to receive(:guess_input_error) }
+      end
+
+      context 'when incorrect register data inputted' do
+        context 'when name is incorrect output message' do
+          before do
+            allow(view_module).to receive(:obtain_name).and_return(wrong_name, name)
+            allow(view_module).to receive(:obtain_difficulty).and_return(difficulty)
+          end
+
+          it { expect(view_module).to receive(:name_length_error) }
+        end
+
+        context 'when difficulty is incorrect output message' do
+          before do
+            allow(view_module).to receive(:obtain_name).and_return(name)
+            allow(view_module).to receive(:obtain_difficulty).and_return(wrong_difficulty, difficulty)
+          end
+
+          it { expect(view_module).to receive(:difficulty_input_error) }
+        end
+      end
+    end
+
+    context 'when hint' do
+      before do
+        allow(view_module).to receive(:fetch_user_input).and_return('start', 'exit')
+        allow(view_module).to receive(:obtain_name).and_return(name)
+        allow(view_module).to receive(:obtain_difficulty).and_return(difficulty)
+      end
+
+      context 'when user enter hint output hint from view' do
+        before { allow(view_module).to receive(:obtain_guess).and_return('hint', 'exit') }
+
+        it { expect(view_module).to receive(:hint) }
+      end
+
+      context 'when hints is used output appropriate message from view' do
+        before { allow(view_module).to receive(:obtain_guess).and_return('hint', 'hint', 'exit') }
+
+        it { expect(view_module).to receive(:no_hints) }
+      end
+    end
+  end
+end
```

Comment:
Please separate and add more specs for game_state, menu_state,
play_state, registration_state, results_state. I didn`t find specs with YAML, stats etc.